sdk/search/azure-search/src/main/java/com/azure/search/models/Field.java

```diff
@@ -0,0 +1,573 @@
+     * handled. Fields of type Edm.String or Collection(Edm.String) that are
+     * filterable do not undergo word-breaking, so comparisons are for exact
+     * matches only. For example, if you set such a field f to "sunny day",
+     * $filter=f eq 'sunny' will find no matches, but $filter=f eq 'sunny day'
+     * will. This property must be null for complex fields. Default is true for
+     * simple fields and null for complex fields.
+     */
+    @JsonProperty(value = "filterable")
+    private Boolean filterable;
+
+    /*
+     * A value indicating whether to enable the field to be referenced in
+     * $orderby expressions. By default Azure Cognitive Search sorts results by
+     * score, but in many experiences users will want to sort by fields in the
+     * documents. A simple field can be sortable only if it is single-valued
+     * (it has a single value in the scope of the parent document). Simple
+     * collection fields cannot be sortable, since they are multi-valued.
+     * Simple sub-fields of complex collections are also multi-valued, and
+     * therefore cannot be sortable. This is true whether it's an immediate
+     * parent field, or an ancestor field, that's the complex collection.
+     * Complex fields cannot be sortable and the sortable property must be null
+     * for such fields. The default for sortable is true for single-valued
+     * simple fields, false for multi-valued simple fields, and null for
+     * complex fields.
+     */
+    @JsonProperty(value = "sortable")
+    private Boolean sortable;
+
+    /*
+     * A value indicating whether to enable the field to be referenced in facet
+     * queries. Typically used in a presentation of search results that
+     * includes hit count by category (for example, search for digital cameras
+     * and see hits by brand, by megapixels, by price, and so on). This
+     * property must be null for complex fields. Fields of type
+     * Edm.GeographyPoint or Collection(Edm.GeographyPoint) cannot be
+     * facetable. Default is true for all other simple fields.
+     */
+    @JsonProperty(value = "facetable")
+    private Boolean facetable;
+
+    /*
+     * The name of the language analyzer to use for the field. This option can
+     * be used only with searchable fields and it can't be set together with
+     * either searchAnalyzer or indexAnalyzer. Once the analyzer is chosen, it
+     * cannot be changed for the field. Must be null for complex fields.
+     */
+    @JsonProperty(value = "analyzer")
+    private String analyzer;
+
+    /*
+     * The name of the analyzer used at search time for the field. This option
+     * can be used only with searchable fields. It must be set together with
+     * indexAnalyzer and it cannot be set together with the analyzer option.
+     * This analyzer can be updated on an existing field. Must be null for
+     * complex fields.
+     */
+    @JsonProperty(value = "searchAnalyzer")
+    private String searchAnalyzer;
```

Comment:
@brjohnstmsft this was a decision made during dev to repalce "extensible enums" with strings. Since in the .net code the unums are repalced with custom code. PR https://github.com/Azure/azure-rest-api-specs/pull/7751 was created but at the time it used extensible enums for only certain classes (TokenFilterName, TokenizerName, CharFilterName). I will send you the discussion seperatly.

Reply:
@navalev We can't rely only on generated code -- we need the API to be user-friendly. In this case, changing it later would be a backwards-incompatible change.

We will find a way to fix this before we ship the SDK. No need to block this PR.

Reply:
added as a task to the backlog maintenance list